kv/kv.go

```diff
@@ -13,6 +13,47 @@
 
 package kv
 
+import "bytes"
+
+// EncodedKey represents encoded key in low-level storage engine.
+type EncodedKey []byte
+
+// Key represents high-level Key type
+type Key []byte
+
+// Next returns the next key in byte-order
+func (k Key) Next() Key {
+	// add \x0 to the end of key
+	return append(append([]byte(nil), []byte(k)...), 0)
```

Comment:
This should allocate slice twice? 

First allocing a slice with same size with k, then appending 0 also re-allocs another slice with size len(k) + 1 ? 

```
buf := make([]byte, len(k) + 1)
copy(buf, k)
return buf
```

maybe more efficient, I don't know.